src/js/lib/monster.ui.js

```diff
@@ -3215,21 +3215,119 @@ define(function(require) {
 		var iconId = args.id;
 		var iconPrefix = iconId.substring(0, iconId.indexOf('--'));
 		var attributes = _.get(args, 'attributes', {});
-		attributes.class = _
-			.chain(attributes)
-			.get('class', '')
-			.split(/\s+/g)      // Split by one or more whitespaces
-			.reject(_.isEmpty)  // Reject empty strings that appear due to leading or trailing whitespaces, or empty string
-			.union(['svg-icon', iconPrefix])
-			.join(' ')
-			.value();
+		attributes = mergeHtmlAttributes(attributes, {
+			'class': 'svg-icon ' + iconPrefix
+		});
 
 		return monster.template(monster.apps.core, 'monster-svg-icon', {
 			iconId: iconId,
 			attributes: attributes
 		});
 	}
 
+	/**
+	 * Generates a key-value pair editor
+	 * @param  {jQuery} $target  Container to append the widget to.
+	 * @param  {Object} [options]  Editor options
+	 * @param  {Object} [options.data]  Key-value data, as a plain object
+	 * @param  {String} [options.inputName]  Input name prefix
+	 * @param  {Object} [options.i18n]  Custom label translations
+	 * @param  {Object} [options.i18n.addLink]  Add row link label
+	 * @param  {String} [options.i18n.keyPlaceholder]  Key input placeholder
+	 * @param  {String} [options.i18n.valuePlaceholder]  Value input placeholder
+	 */
+	function keyValueEditor($target, options) {
+		if (!($target instanceof $)) {
+			throw TypeError('"$target" is not a jQuery object');
+		}
+		var data = _.get(options, 'data', {});
+		var inputName = _.get(options, 'inputName', 'data');
+		var addLink = _.get(options, 'i18n.addLink');
+		var keyPlaceholder = _.get(options, 'i18n.keyPlaceholder');
+		var valuePlaceholder = _.get(options, 'i18n.valuePlaceholder');
+		if (!_.isPlainObject(data)) {
+			throw new TypeError('"options.data" is not a plain object');
+		}
+		if (!_.isNil(inputName) && !_.isString(inputName)) {
+			throw new TypeError('"options.inputName" is not a string');
+		}
+		if (!_.isNil(addLink) && !_.isString(addLink)) {
+			throw new TypeError('"options.i18n.addLink" is not a string');
+		}
+		if (!_.isNil(keyPlaceholder) && !_.isString(keyPlaceholder)) {
+			throw new TypeError('"options.i18n.keyPlaceholder" is not a string');
+		}
+		if (!_.isNil(valuePlaceholder) && !_.isString(valuePlaceholder)) {
+			throw new TypeError('"options.i18n.valuePlaceholder" is not a string');
+		}
+		var $editorTemplate = $(monster.template(monster.apps.core, 'monster-key-value-editor', {
+			addLink: addLink
+		}));
+		var $rowContainer = $editorTemplate.find('.monster-key-value-data-container');
+		var addRow = function(value, key, index) {
+			$rowContainer.append(monster.template(monster.apps.core, 'monster-key-value-editor-row', {
+				key: key,
+				value: value,
+				inputName: inputName + '[' + index + ']',
+				keyPlaceholder: keyPlaceholder,
+				valuePlaceholder: valuePlaceholder
+			}));
+		};
+		// Add initial rows
+		var counter = 0;
+		_.each(data, function(value, key, index) {
+			addRow(value, key, counter++);
```

Comment:
Using unary increment/decrement operators is usually not a good idea as it can hide unintentional effects leading to unexpected behaviors.

It is more expressive to increment/decrement variables by using statements:
```suggestion
			addRow(value, key, counter);
			counter += 1;
```